crates/kubelet/src/volumes.rs

```diff
@@ -0,0 +1,186 @@
+//! A module for use in managing volumes in providers. Use of this module is not mandatory to create
+//! a Provider, but it does provide common implementation logic for supported volume providers.
+use std::collections::HashMap;
+use std::ops::Deref;
+use std::path::PathBuf;
+
+use k8s_openapi::api::core::v1::Volume as KubeVolume;
+use k8s_openapi::api::core::v1::{ConfigMap, Secret};
+use kube::api::Api;
+use log::{debug, error};
+
+use crate::Pod;
+
+#[derive(Debug)]
+enum VolumeType {
+    ConfigMap,
+    Secret,
+    HostPath,
+}
+
+/// A smart wrapper around the location of a volume on the host system. If this is a ConfigMap or
+/// Secret volume, dropping this reference will clean up the temporary volume. [AsRef] and
+/// [std::ops::Deref] are implemented for this type so you can still use it like a normal PathBuf
+pub struct VolumeRef {
+    host_path: PathBuf,
+    volume_type: VolumeType,
+}
+
+impl VolumeRef {
+    /// Resolves the volumes for a pod, including preparing temporary directories containing the
+    /// contents of secrets and configmaps. Returns a HashMap of volume names to a PathBuf for the
+    /// directory where the volume is mounted
+    pub async fn volumes_from_pod(
+        volume_dir: &PathBuf,
+        pod: &Pod,
+        client: &kube::Client,
+    ) -> anyhow::Result<HashMap<String, Self>> {
+        let base_path = volume_dir.join(pod_dir_name(pod));
```

Comment:
this appears to be the only location where `pod_dir_name` is called, and it just looks to be a `format!` call. Is there a particular reason this was broken out as a separate function?

Would it make sense to make the string format (`"{}-{}"`) a constant instead, or is this being used as a constant reference later down the road for something?

Reply:
I'll chalk it up to premature optimization. I figured it could possibly be used more down the road. Also, I like that it gives a human readable name to the functionality instead of a bare format call